tree/ntuple/v7/test/rfield_class.cxx

```diff
@@ -18,3 +26,107 @@ TEST(RNTuple, TClass) {
    FileRaii fileGuard("test_ntuple_tclass.root");
    auto ntuple = RNTupleWriter::Recreate(std::move(model), "f", fileGuard.GetPath());
 }
+
+TEST(RTNuple, TObject)
+{
+   EXPECT_THROW(std::make_unique<ROOT::Experimental::RClassField>("obj", "TObject"), RException);
+
+   FileRaii fileGuard("test_ntuple_tobject.root");
+   {
+      auto model = RNTupleModel::Create();
+      model->MakeField<TObject>("obj");
+      auto writer = RNTupleWriter::Recreate(std::move(model), "ntpl", fileGuard.GetPath());
+      auto entry = writer->GetModel().CreateBareEntry();
+
+      auto heapObj = std::make_unique<TObject>();
+      EXPECT_TRUE(heapObj->TestBit(TObject::kIsOnHeap));
+      EXPECT_TRUE(heapObj->TestBit(TObject::kNotDeleted));
+      heapObj->SetUniqueID(137);
+      entry->BindRawPtr("obj", heapObj.get());
+      writer->Fill(*entry);
+
+      // Saving a destructed object is here to verify that the RNTuple serialization does the same than
+      // the TObject custom streamer (i.e., ignoring the kNotDeleted flag)
+      heapObj->~TObject();
+      EXPECT_FALSE(heapObj->TestBit(TObject::kNotDeleted));
+      writer->Fill(*entry);
+
+      TObject stackObj;
+      EXPECT_FALSE(stackObj.TestBit(TObject::kIsOnHeap));
+      entry->BindRawPtr("obj", &stackObj);
+      writer->Fill(*entry);
+
+      stackObj.SetBit(TObject::kIsReferenced);
+      EXPECT_THROW(writer->Fill(*entry), RException);
+   }
+
+   auto reader = RNTupleReader::Open("ntpl", fileGuard.GetPath());
+   EXPECT_EQ(3u, reader->GetNEntries());
+
+   auto entry = reader->GetModel().CreateBareEntry();
+   TObject stackObj;
+   entry->BindRawPtr("obj", &stackObj);
+
+   reader->LoadEntry(0, *entry);
+   EXPECT_EQ(137u, stackObj.GetUniqueID());
+   EXPECT_FALSE(stackObj.TestBit(TObject::kIsOnHeap));
+   EXPECT_TRUE(stackObj.TestBit(TObject::kNotDeleted));
+
+   reader->LoadEntry(1, *entry);
+   EXPECT_EQ(137u, stackObj.GetUniqueID());
+   EXPECT_FALSE(stackObj.TestBit(TObject::kIsOnHeap));
+   EXPECT_TRUE(stackObj.TestBit(TObject::kNotDeleted));
+
+   auto heapObj = std::make_unique<TObject>();
+   entry->BindRawPtr("obj", heapObj.get());
+   reader->LoadEntry(2, *entry);
+   EXPECT_EQ(0u, heapObj->GetUniqueID());
+   EXPECT_TRUE(heapObj->TestBit(TObject::kIsOnHeap));
+   EXPECT_TRUE(heapObj->TestBit(TObject::kNotDeleted));
+
+   heapObj->SetBit(TObject::kIsReferenced);
+   EXPECT_THROW(reader->LoadEntry(2, *entry), RException);
```

Comment:
Also this part seems like it's testing some other functionality, maybe a comment or a separate test